libethcore/Farm.h

```diff
@@ -64,8 +64,11 @@ class Farm : public FarmFace
         std::function<Miner*(FarmFace&, unsigned)> create;
     };
 
-    Farm(boost::asio::io_service& io_service) : m_io_strand(io_service), m_hashrateTimer(io_service)
+    Farm(boost::asio::io_service& io_service, bool hwmon, bool pwron) : m_io_strand(io_service), m_collectTimer(io_service)
     {
+        m_hwmon = hwmon;
```

Comment:
since you added ``hwmon`` on creation of farm, you could put the initialisation of the HWMON classes within an if (hwmon) too.

Reply:
With your suggestion we might trim something in memory usage and init time.
Nevertheless we might implement in future an API method to enable/disable hardware monitoring